python/ml4ir/base/model/architectures/auto_dag_network.py

```diff
@@ -274,6 +329,37 @@ def instantiate_op(self, current_layer_type: str, current_layer_args: Dict) -> l
             raise KeyError(f"Layer type: '{current_layer_type}' "
                            f"is not supported or not found in subclasses of keras.layers.Layer")
 
+    def get_layer_op(
+            self,
+            layer_args: Dict,
+            existing_op: layers.Layer = None
+    ) -> Dict[str, Union[str, bool, layers.Layer]]:
+        """
+        Define the layer operation for a layer in the model config
+
+        Parameters
+        ----------
+        layer_args: dict
+            All the arguments from model config which are needed to define a layer op
+        existing_op: instance of type keras.layers.Layer
+            If specified, reuse the layer op
+
+        Returns
+        -------
+        dict
+            Dictionary with the layer instance, inputs required for the layer and the format of inputs to the layer
+        """
+        # Ensure the class names for the current layer and the existing op are the same
+        if existing_op and existing_op.__class__.__name__ != layer_args[self.LAYER_TYPE].split(".")[-1]:
+            raise TypeError(f"Cannot reuse existing layer of type {existing_op.__class__.__name__}. "
+                            f"{layer_args[self.LAYER_TYPE]} is incompatible")
+        return {
+            self.INPUTS: layer_args[self.INPUTS],
+            self.OP_IDENTIFIER: existing_op if existing_op
```

Comment:
@arvindsrikantan So tying weights is just a matter of using the same layer? I assumed there would be two different layers sharing the tensorflow variables somehow. Glad that this seems to be much more straightforward.

Reply:
This feels very RNN backprop like. Correct me if my understanding is wrong.

Reply:
Yeah, pretty much like an RNN. In this case it boils down to using the same layer. We currently don't support word2vec type weight tying